File: 0844-backspace-string-compare/0844-backspace-string-compare.go
func backspaceCompare(s string, t string) bool {
    var hash byte = 35
    sStack, tStack := []byte{}, []byte{}

    for i := 0; i < len(s); i++ {
        if s[i] == hash {
            if len(sStack) > 0 {
                sStack = sStack[:len(sStack) - 1]
            }
        } else {
            sStack = append(sStack, s[i])
        }
    }

    for i := 0; i < len(t); i++ {
        if t[i] == hash {
            if len(tStack) > 0 {
                tStack = tStack[:len(tStack) - 1]
            }
        } else {
            tStack = append(tStack, t[i])
        }
    }
    
    return string(sStack) == string(tStack)
}

File: 0844-backspace-string-compare/0844-backspace-string-compare.cpp
static const bool a = [](){
    std::ios_base::sync_with_stdio(false);
    std::cout.tie(nullptr);
    std::cin.tie(nullptr);
    return true;
}();
class Solution {
public:
    bool backspaceCompare(const string s, const string t) {
        stack<char> sStack;
        stack<char> tStack;
        for (const char c : s) {
            if (c == '#') {
                if (!sStack.empty()) {
                    sStack.pop();
                }
            } else {
                sStack.push(c);
            }
        } 
        for (const char c : t) {
            if (c == '#') {
                if (!tStack.empty()) {
                    tStack.pop();
                }
            } else {
                tStack.push(c);
            }
        }
        return sStack == tStack;
    }
};